python_modules/libraries/dagster-dg-cli/dagster_dg_cli/cli/scaffold/branch.py

```diff
@@ -318,3 +358,22 @@ def scaffold_branch_command(
     pr_url = push_branch_and_create_pr(branch_name, pr_title, pr_body)
 
     click.echo(f"✅ Successfully created branch and pull request: {pr_url}")
+
+    if record:
+        session_id = uuid.uuid4()
+        session_data = {
+            "timestamp": datetime.now().isoformat(),
+            "dg_version": _get_dg_version(),
+            "branch_name": branch_name,
+            "pr_title": pr_title,
+            "pr_url": pr_url,
+            "branch_base": branch_base,
+            "input": {
+                "prompt": prompt_text,
+            },
+            "output": generated_outputs,
+        }
```

Comment:
Can we please put this in a @record or something for documentation and type safety

Reply:
As well as some commentary at how one would reconstruct the actual code generated from this information.

Reply:
believe the updates address this, let me know